docs/spec/slashing/state-machine.md

```diff
@@ -0,0 +1,174 @@
+*V<sub>b</sub>* : validator bonded  
+*V<sub>u</sub>* : validator unbonded  
+
+*Single infraction*
+
+<----------------->   
+[----------C<sub>1</sub>----D<sub>1</sub>,V<sub>u</sub>-----]
+
+A single infraction is committed then later discovered, at which point the validator is unbonded and slashed at the full amount for the infraction.
+
+*Multiple infractions*
+
+<---------------------------->   
+[----------C<sub>1</sub>--C<sub>2</sub>---C<sub>3</sub>---D<sub>1</sub>,D<sub>2</sub>,D<sub>3</sub>V<sub>u</sub>-----]
+
+Multiple infractions are committed within a single slashing period then later discovered, at which point the validator is unbonded and slashed for only the worst infraction.
+
+*Multiple infractions after rebonding*
+
+
+<---------------------------->&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<-------------->  
+[----------C<sub>1</sub>--C<sub>2</sub>---C<sub>3</sub>---D<sub>1</sub>,D<sub>2</sub>,D<sub>3</sub>V<sub>u</sub>---V<sub>b</sub>---C<sub>4</sub>----D<sub>4</sub>,V<sub>u</sub>--]
+
+Multiple infractions are committed within a single slashing period then later discovered, at which point the validator is unbonded and slashed for only the worst infraction.
+The validator then unjails themself and rebonds, then commits a fourth infraction - which is discovered and punished at the full amount, since a new slashing period started
+when they unjailed and rebonded.
+
+## Transactions
+
+In this section we describe the processing of transactions for the `slashing` module.
+
+### TxUnjail
+
+If a validator was automatically unbonded due to downtime and wishes to come back online &
+possibly rejoin the bonded set, it must send `TxUnjail`:
+
+```
+type TxUnjail struct {
+    ValidatorAddr sdk.AccAddress
+}
+
+handleMsgUnjail(tx TxUnjail)
+
+    validator = getValidator(tx.ValidatorAddr)
+    if validator is nil
+      fail with "No validator found"
+
+    if validator is not jailed
+      fail with "Validator not jailed, cannot unjail"
+
+    info = getValidatorSigningInfo(operator)
+    if block time is before info.JailedUntil
+      fail with "Validator still jailed, cannot unjail until period has expired"
+
+    // Update the start height so the validator won't be immediately unbonded again
+    info.StartHeight = BlockHeight
+    setValidatorSigningInfo(info)
+
+    validator.Jailed = false
+    setValidator(validator)
+
+    return
+```
+
+If the validator has enough stake to be in the top hundred, they will be automatically rebonded,
+and all delegators still delegated to the validator will be rebonded and begin to again collect
+provisions and rewards.
+
+## Interactions
+
+In this section we describe the "hooks" - slashing module code that runs when other events happen.
+
+### Validator Bonded
+
+Upon successful bonding of a validator (a given validator changing from "unbonded" state to "bonded" state,
```

Comment:
`changing from "unbonded" state to "bonded" state`, not necessarily true, can also be coming from `unbonding` state. I'd just say "entering the bonded state"

Reply:
The problem is that we have two terms: "unbonded" == "not bonded", and "unbonded == sdk.Unbonded state", because we call the state itself unbonded it's unclear what to refer to the former by.

I like your wording better anyways but maybe we should consider renaming the state.

Reply:
I disagree with `"unbonded" == "not bonded"` I think by definition:
`"unbonded" || "unbonding" == "not bonded"`

Reply:
Right, but "unbonded" is also an English *word*, e.g. https://www.dictionary.com/browse/unbonded, which means "not bonded" (is the antonym of "bonded") - and I presume this etymology is the same root as for Cosmos; e.g. https://en.wikipedia.org/wiki/Bonded_warehouse (perhaps)?

What do you think we should call the state of "not bonded", if not "unbonded"?

Reply:
(I don't use "unbonded" this way in the spec anymore though)

Reply:
okay - yeah confusing. We should probably just completely avoid the language unbonded and only refer to things AS bonded/unbonding/unbonded - and define this strictly as meaning one of the 3 states of tokens. 

However I will draw the subtle generalized distinction (as I understand it) between the prefix "un-" and the the adverb "not". The word "not" is rooted in the idea of exclusion "exclude a person or part of a group", whereas "un-" when used as `prefix forming verbs and verbal derivatives` (unbonding = a state = a verb) is used to refer to the deviation or movement from that original state. SO in this context I would sustain that the state which is excluding the bonded state is unique from the state which is the deviation from the bonded state - aka 
`superset(unbonded, unbonding) == not bonded` 
`subset(not bonded) == unbonded || unbonding`

"subtle" lol